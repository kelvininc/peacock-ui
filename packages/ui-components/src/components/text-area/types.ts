import { EventEmitter } from '@stencil/core';
import { EIconName, EOtherIconName } from '../icon/icon.types';

export interface ITextArea {
	/** (optional) Icon to show to the left of the text field */
	icon?: EIconName | EOtherIconName;
	/** (optional) The text to show inside the text area */
	text?: string;
	/** (optional) The placeholder to show in the text area */
	placeholder?: string;
	/** (required) The maximum number of characters allowed */
	maxCharLength: number;
}

export interface ITextAreaEvents {
	/** Emits the current text when there's a change */
	textChange: EventEmitter<string>;
	/** Emits the current text when the text area loses focus */
	textChangeBlur: EventEmitter<string>;
}
