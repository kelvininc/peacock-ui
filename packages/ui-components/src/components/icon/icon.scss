@import '../../assets/styles/globals';

:host {
	/**
	 * @prop --icon-color: Color of the icon
	 * @prop --icon-height: Height of icon
	 * @prop --icon-width: Width of icon
	 * @prop --icon-rotation: Rotation of icon
	 */
	--icon-color: #{kv-color('text')};
	--icon-height: 16px;
	--icon-width: 16px;
	--icon-rotation: 0deg;
}

.icon {
	fill: var(--icon-color);
	width: var(--icon-width);
	height: var(--icon-height);
	transform: rotate(var(--icon-rotation));

	/* SIZES */
	&.icon-8 {
		width: 8px;
		height: 8px;
	}

	&.icon-12 {
		width: 12px;
		height: 12px;
	}

	&.icon-16 {
		width: 16px;
		height: 16px;
	}

	&.icon-20 {
		width: 20px;
		height: 20px;
	}

	&.icon-24 {
		width: 24px;
		height: 24px;
	}

	&.icon-40 {
		width: 40px;
		height: 40px;
	}

	&.icon-full-size {
		width: 100%;
		height: 100%;
	}

	/* ROTATIONS */
	&.rotate-0 {
		transform: rotate(0deg);
	}

	&.rotate-45 {
		transform: rotate(45deg);
	}

	&.rotate-90 {
		transform: rotate(90deg);
	}

	&.rotate-135 {
		transform: rotate(135deg);
	}

	&.rotate-180 {
		transform: rotate(180deg);
	}

	&.rotate-225 {
		transform: rotate(225deg);
	}

	&.rotate-270 {
		transform: rotate(270deg);
	}

	&.rotate-315 {
		transform: rotate(315deg);
	}
}
