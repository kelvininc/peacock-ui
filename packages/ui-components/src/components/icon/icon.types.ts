export enum EIconName {
	Add = 'kv-add',
	AddCrossCircle = 'kv-add-cross-circle',
	AlarmClass = 'kv-alarm-class',
	Alert = 'kv-alert',
	AlignBottom = 'kv-align-bottom',
	AlignTop = 'kv-align-top',
	Application = 'kv-application',
	Array = 'kv-array',
	ArrowDownload = 'kv-arrow-download',
	ArrowDropDown = 'kv-arrow-drop-down',
	ArrowDropUp = 'kv-arrow-drop-up',
	ArrowLeft = 'kv-arrow-left',
	ArrowRight = 'kv-arrow-right',
	Asset = 'kv-asset',
	AssetA = 'kv-asset-a',
	AssetAStatus = 'kv-asset-a-status',
	AssetC = 'kv-asset-c',
	AssetClass = 'kv-asset-class',
	AssetCStatus = 'kv-asset-c-status',
	AssetP = 'kv-asset-p',
	AssetPStatus = 'kv-asset-p-status',
	AssetS = 'kv-asset-s',
	AssetSStatus = 'kv-asset-s-status',
	AssetStatus = 'kv-asset-status',
	AssetType = 'kv-asset-type',
	Assistant = 'kv-assistant',
	Auditlogs = 'kv-auditlogs',
	Back = 'kv-back',
	BubbleChart = 'kv-bubble-chart',
	Building = 'kv-building',
	Bullet = 'kv-bullet',
	Calendar = 'kv-calendar',
	CalendarUpdate = 'kv-calendar-update',
	Card = 'kv-card',
	Chat = 'kv-chat',
	Circle = 'kv-circle',
	Clear = 'kv-clear',
	Close = 'kv-close',
	CloseCircle = 'kv-close-circle',
	Cloud = 'kv-cloud',
	Code = 'kv-code',
	Collapse = 'kv-collapse',
	Connection = 'kv-connection',
	Copy = 'kv-copy',
	CopyClipboard = 'kv-copy-clipboard',
	Column = 'kv-column',
	Cpu = 'kv-cpu',
	Dashboard = 'kv-dashboard',
	DashboardCustom = 'kv-dashboard-custom',
	Datalabel = 'kv-datalabel',
	Delete = 'kv-delete',
	DensityLow = 'kv-density-low',
	DensityMedium = 'kv-density-medium',
	DensityHigh = 'kv-density-high',
	Device = 'kv-device',
	DeviceType = 'kv-device-type',
	Diamond = 'kv-diamond',
	Docs = 'kv-docs',
	DoubleArrowLeft = 'kv-double-arrow-left',
	DoubleArrowRight = 'kv-double-arrow-right',
	DoneAll = 'kv-done-all',
	DoNotDisturbOnCircle = 'kv-do-not-disturb-on-circle',
	Download = 'kv-download',
	DragDrop = 'kv-drag-drop',
	Edit = 'kv-edit',
	EntityArea = 'kv-entity-area',
	EntityAsset = 'kv-entity-asset',
	EntityComponent = 'kv-entity-component',
	EntityDefault = 'kv-entity-default',
	EntityEnterprise = 'kv-entity-enterprise',
	EntityPart = 'kv-entity-part',
	EntityProcCell = 'kv-entity-proc-cell',
	EntityProdLine = 'kv-entity-prod-line',
	EntityProdUnit = 'kv-entity-prod-unit',
	EntitySensor = 'kv-entity-sensor',
	EntitySite = 'kv-entity-site',
	EntityStorage = 'kv-entity-storage',
	Error = 'kv-error',
	Exit = 'kv-exit',
	Expand = 'kv-expand',
	ExternalLink = 'kv-external-link',
	Eye = 'kv-eye',
	EyeClosed = 'kv-eye-closed',
	Failure = 'kv-failure',
	File = 'kv-file',
	Filter = 'kv-filter',
	Five = 'kv-five',
	Fixed = 'kv-fixed',
	Folder = 'kv-folder',
	Four = 'kv-four',
	Fullscreen = 'kv-fullscreen',
	FullscreenExit = 'kv-fullscreen-exit',
	Gateway = 'kv-gateway',
	Gear = 'kv-gear',
	In = 'kv-in',
	Info = 'kv-info',
	Layer = 'kv-layer',
	LegoLetter = 'kv-lego-letter',
	Link = 'kv-link',
	LinkOff = 'kv-link-off',
	List = 'kv-list',
	Load = 'kv-load',
	Lock = 'kv-lock',
	LogoKelvin = 'kv-logo-kelvin',
	Magictip = 'kv-magictip',
	Maps = 'kv-maps',
	MapsEditor = 'kv-maps-editor',
	Memory = 'kv-memory',
	MenuCollapse = 'kv-menu-collapse',
	MenuExpand = 'kv-menu-expand',
	Metrics = 'kv-metrics',
	Minus = 'kv-minus',
	More = 'kv-more',
	Multiline = 'kv-multiline',
	Nabla = 'kv-nabla',
	NavClose = 'kv-nav-close',
	NavOpen = 'kv-nav-open',
	Notifications = 'kv-notifications',
	One = 'kv-one',
	Out = 'kv-out',
	Pause = 'kv-pause',
	Place = 'kv-place',
	Play = 'kv-play',
	Protip = 'kv-protip',
	Refresh = 'kv-refresh',
	ResetZoom = 'kv-reset-zoom',
	Restart = 'kv-restart',
	Reset = 'kv-reset',
	Rotate = 'kv-rotate',
	Save = 'kv-save',
	Search = 'kv-search',
	Settings = 'kv-settings',
	Share = 'kv-share',
	SlimRight = 'kv-slim-right',
	SortAz = 'kv-sort-az',
	SortHover = 'kv-sort-hover',
	SortZa = 'kv-sort-za',
	Square = 'kv-square',
	Status = 'kv-status',
	Star = 'kv-star',
	Success = 'kv-success',
	SupervisedUser = 'kv-supervised-user',
	SystemSource = 'kv-system-source',
	Three = 'kv-three',
	Time = 'kv-time',
	Todo = 'kv-todo',
	Tree = 'kv-tree',
	TreeCollapse = 'kv-tree-collapse',
	TreeExpand = 'kv-tree-expand',
	TrendingDown = 'kv-trending-down',
	TrendingUp = 'kv-trending-up',
	Triangle = 'kv-triangle',
	Two = 'kv-two',
	Type = 'kv-type',
	UnfoldMore = 'kv-unfold-more',
	Upgrade = 'kv-upgrade',
	Upload = 'kv-upload',
	Uptime = 'kv-uptime',
	Warning = 'kv-warning',
	ZoomIn = 'kv-zoom-in',
	ZoomOut = 'kv-zoom-out'
}

export enum EOtherIconName {
	IconActivity = 'icon-activity',
	IconField = 'icon-field',
	IconModel = 'icon-model',
	IconGear = 'icon-gear',
	IconControl = 'icon-control',
	IconNotes = 'icon-notes',
	IconSearch = 'icon-search',
	IconKelvin = 'icon-kelvin',
	IconChart = 'icon-chart',
	IconTrash = 'icon-trash',
	IconClearCross = 'icon-clear-cross',
	IconPaperclip = 'icon-paperclip',
	IconShadowMismatch = 'icon-shadow-mismatch',
	IconCollapseOpen = 'icon-collapse-open',
	IconCollapseClose = 'icon-collapse-close',
	IconRadioBtn = 'icon-radio-btn',
	IconRadioBtnSelected = 'icon-radio-btn-selected',
	IconCheckmark = 'icon-checkmark',
	IconArrow = 'icon-arrow',
	IconLock = 'icon-lock',
	IconUser = 'icon-user',
	IconAddCross = 'icon-add-cross',
	IconSwitchEnabled = 'icon-switch-enabled',
	IconSwitchDisabled = 'icon-switch-disabled',
	IconPencil = 'icon-pencil',
	IconGcp = 'icon-gcp',
	IconKafka = 'icon-kafka',
	IconAzure = 'icon-azure',
	IconSortArrow = 'icon-sort-arrow',
	IconSortUnset = 'icon-sort-unset',
	IconClearCrossCircle = 'icon-clear-cross-circle',
	IconCheckState = 'icon-check-state',
	IconCheckHoverState = 'icon-check-hover-state',
	IconUncheckState = 'icon-uncheck-state',
	IconIndeterminateState = 'icon-indeterminate-state',
	IconLeftArrowThin = 'icon-left-arrow-thin',
	IconRightArrowThin = 'icon-right-arrow-thin',
	IconSaveDisk = 'icon-save-disk',
	IconDashboards = 'icon-dashboards',
	IconDragHorizontal = 'icon-drag-horizontal',
	IconDragVertical = 'icon-drag-vertical',
	IconEntity = 'icon-entity',
	IconGroup = 'icon-group',
	IconFracVan = 'icon-frac-van',
	IconOverview = 'icon-overview',
	IconWireline = 'icon-wireline',
	IconInbox = 'icon-inbox',
	IconSetup = 'icon-setup',
	IconTeam = 'icon-team',
	IconWell = 'icon-well',
	IconComposite = 'icon-composite',
	IconDragCursor = 'icon-drag-cursor',
	IconInfo = 'icon-info',
	IconPumpSchedule = 'icon-pump-schedule',
	IconCancel = 'icon-cancel',
	IconWarning = 'icon-warning',
	IconCalendarMinus = 'icon-calendar-minus',
	IconCalendarPlus = 'icon-calendar-plus',
	IconMore = 'icon-more',
	IconRedo = 'icon-redo',
	IconUndo = 'icon-undo',
	IconCopy = 'icon-copy',
	IconLog = 'icon-log',
	IconMapPoint = 'icon-map-point',
	IconStackedBarChart = 'icon-stacked-bar-chart',
	IconTimelineRangeRight = 'icon-timeline-range-right',
	IconTimelineRangeKeft = 'icon-timeline-range-left'
}
