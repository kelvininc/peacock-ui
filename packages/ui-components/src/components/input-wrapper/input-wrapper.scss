@import '../../assets/styles/globals';

:host {
	display: block;
}

$input-height: 32px;

.input-container {
	background-color: kv-color('neutral-7');
	padding: 0 $spacing-4x;
	height: $input-height;
	display: flex;
	align-items: center;

	kv-icon {
		visibility: hidden;
	}

	&--content-hidden {
		&:hover {
			background-color: kv-color('neutral-6');
			cursor: pointer;

			kv-icon {
				visibility: visible;
			}
		}
	}

	.label-container {
		@include kv-font-label-small-regular;
		display: flex;
		align-items: center;
		justify-content: space-between;
		width: 100%;

		color: kv-color('neutral-4');
	}

	.slot-container {
		width: 100%;
	}
}
