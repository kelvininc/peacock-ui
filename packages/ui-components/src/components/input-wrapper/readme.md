# kv-time-picker-timezone



<!-- Auto Generated Below -->


## Usage

### Angular

```html
<!-- Default -->
<kv-input-wrapper contentVisible>
	<kv-text-field label="Text Field"></kv-text-field>
</kv-input-wrapper>

<!-- With Label -->
<kv-input-wrapper label="Content Label">
	<kv-text-field label="Text Field"></kv-text-field>
</kv-input-wrapper>

<!-- With component hidden -->
<kv-input-wrapper label="Content Label" contentVisible={false}>
	<kv-text-field label="Text Field"></kv-text-field>
</kv-input-wrapper>
```


### React

```tsx
import React from 'react';
import { KvInputWrapper, KvTextField } from '@kelvininc/react-ui-components';

export const KvInputWrapperExample: React.FC = () => (
  <>
	<KvInputWrapper label="Options" contentVisible={false}>
		<KvTextField label="label" />
	</KvInputWrapper>
  </>
);
```



## Properties

| Property                      | Attribute          | Description                                                                  | Type        | Default          |
| ----------------------------- | ------------------ | ---------------------------------------------------------------------------- | ----------- | ---------------- |
| `contentDisabled`             | `content-disabled` | Defines if the content can be seen                                           | `boolean`   | `false`          |
| `contentVisible` _(required)_ | `content-visible`  | Defines the state of the component                                           | `boolean`   | `undefined`      |
| `icon`                        | `icon`             | Icon that is displayed when the component is hovered and is in default state | `EIconName` | `EIconName.Edit` |
| `label`                       | `label`            | Label that is shown when the component is in Default state                   | `string`    | `undefined`      |


## Events

| Event          | Description                                                                | Type                   |
| -------------- | -------------------------------------------------------------------------- | ---------------------- |
| `contentClick` | Event emitted when the component is clicked to display the wrapped content | `CustomEvent<boolean>` |


## Dependencies

### Used by

 - [kv-relative-time-picker](../relative-time-picker)

### Depends on

- [kv-icon](../icon)

### Graph
```mermaid
graph TD;
  kv-input-wrapper --> kv-icon
  kv-relative-time-picker --> kv-input-wrapper
  style kv-input-wrapper fill:#f9f,stroke:#333,stroke-width:4px
```

----------------------------------------------


