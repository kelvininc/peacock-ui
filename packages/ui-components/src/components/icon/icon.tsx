import { Component, Prop, Host, h } from '@stencil/core';
import { getClassMap } from '../../utils/css-class.helper';
import { getConfig } from '../utils';
import { EIconName, EOtherIconName } from './icon.types';

/**
 * @part icon - The icon container.
 */
@Component({
	tag: 'kv-icon',
	styleUrl: 'icon.scss',
	shadow: true
})
export class KvIcon {
	/** (required) Icon symbol name */
	@Prop({ reflect: true }) name!: EIconName | EOtherIconName;

	/**
	 * (optional) Additional classes to apply for custom CSS. If multiple classes are
	 * provided they should be separated by spaces.
	 */
	@Prop({ reflect: true }) customClass: string | string[] = '';

	/** (optional) Icon custom color */
	@Prop({ reflect: true }) customColor: string = '';

	render() {
		const { baseAssetsUrl } = getConfig();

		return (
			<Host>
				<svg
					part="icon"
					xmlns="http://www.w3.org/2000/svg"
					class={{
						...getClassMap(this.customClass),
						icon: true
					}}
					style={{ fill: this.customColor }}
				>
					<use href={`${baseAssetsUrl}svg-symbols.svg#${this.name}`}></use>
				</svg>
			</Host>
		);
	}
}
