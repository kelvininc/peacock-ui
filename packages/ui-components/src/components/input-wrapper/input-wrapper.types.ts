import { EventEmitter } from '@stencil/core';
import { EIconName } from '../icon/icon.types';

export interface IInputWrapper {
	/** Defines the state of the component */
	contentVisible: boolean;
	/** Label that is shown when the component is in Default state */
	label?: string;
	/** Icon that is displayed when the component is hovered and is in default state */
	icon?: EIconName;
}

export interface IInputWrapperEvents {
	/** Event emitted when the component is clicked to display the wrapped content*/
	contentClick: EventEmitter<boolean>;
}
