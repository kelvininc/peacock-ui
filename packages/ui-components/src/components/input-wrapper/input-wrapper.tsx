import { Component, Event, EventEmitter, Host, Prop, h } from '@stencil/core';
import { IInputWrapper, IInputWrapperEvents } from './input-wrapper.types';
import { EIconName } from '../icon/icon.types';

@Component({
	tag: 'kv-input-wrapper',
	styleUrl: 'input-wrapper.scss',
	shadow: true
})
export class KvInputWrapper implements IInputWrapper, IInputWrapperEvents {
	/** @inheritdoc */
	@Prop({ reflect: false }) contentVisible!: boolean;
	/** @inheritdoc */
	@Prop({ reflect: false }) contentDisabled?: boolean = false;
	/** @inheritdoc */
	@Prop({ reflect: false }) label?: string;
	/** @inheritdoc */
	@Prop({ reflect: false }) icon?: EIconName = EIconName.Edit;

	/** @inheritdoc */
	@Event() contentClick: EventEmitter<boolean>;

	private handleContentClick = () => {
		if (!this.contentVisible && !this.contentDisabled) {
			this.contentClick.emit(this.contentVisible);
		}
	};

	render() {
		return (
			<Host>
				<div
					class={{
						'input-container': true,
						'input-container--content-hidden': !this.contentVisible && !this.contentDisabled
					}}
					onClick={this.handleContentClick}
				>
					{this.contentVisible && !this.contentDisabled ? (
						<div class="slot-container">
							<slot />
						</div>
					) : (
						<div class="label-container">
							<div class="label">{this.label}</div>
							<kv-icon name={this.icon} />
						</div>
					)}
				</div>
			</Host>
		);
	}
}
