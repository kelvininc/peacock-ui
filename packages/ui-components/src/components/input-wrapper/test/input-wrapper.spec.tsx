import { SpecPage, newSpecPage } from '@stencil/core/testing';
import { KvInputWrapper } from '../input-wrapper';
import { h } from '@stencil/core';
import { EIconName } from '../../icon/icon.types';

describe('KvInputWrapper (unit tests)', () => {
	let page: SpecPage;
	let component: KvInputWrapper;

	beforeEach(async () => {
		page = await newSpecPage({
			components: [KvInputWrapper],
			template: () => <kv-input-wrapper label="Label" contentVisible={false}></kv-input-wrapper>
		});
		component = page.rootInstance;
	});

	describe('when the component loads', () => {
		it('should match the snapshot', () => {
			expect(page.root).toMatchSnapshot();
		});

		it('should set the correct label', () => {
			expect(component.label).toEqual('Label');
		});

		it('should set the correct icon', () => {
			expect(component.icon).toEqual(EIconName.Edit);
		});

		it('should set the correct contentVisible setting', () => {
			expect(component.contentVisible).toEqual(false);
		});
	});
});
