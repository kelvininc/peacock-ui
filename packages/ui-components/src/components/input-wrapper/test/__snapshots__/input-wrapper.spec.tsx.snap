// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`KvInputWrapper (unit tests) when the component loads should match the snapshot 1`] = `
<kv-input-wrapper>
  <mock:shadow-root>
    <div class="input-container input-container--content-hidden">
      <div class="label-container">
        <div class="label">
          Label
        </div>
        <kv-icon name="kv-edit"></kv-icon>
      </div>
    </div>
  </mock:shadow-root>
</kv-input-wrapper>
`;
